resources/providers/awslib/ec2/ec2_instance.go

```diff
@@ -48,6 +49,14 @@ func (i Ec2Instance) GetResourceName() string {
 	return ""
 }
 
+func (i Ec2Instance) GetResourceId() string {
+	if i.Instance.InstanceId == nil {
+		return ""
+	}
+
+	return strings.Split(*i.Instance.InstanceId, "-")[1]
```

Comment:
Why remove the `id` prefix?

Reply:
The resource id shouldn't be coupled with a solution.
I think that having a hash is more future-proof than having the `i-` prefix that isn't unique and doesn't contribute anything to the uniqueness and just vendor locks us to the AWS representation.